client/blocks/product-selector/index.jsx

```diff
@@ -307,6 +316,11 @@ export class ProductSelector extends Component {
 		) {
 			upsellProductSlug = product.productUpsells[ productPurchase.productSlug ];
 
+			// If we have yearly plan, don't show yearly upsell in monthly view.
+			if ( 'monthly' === intervalType && productPriceMatrix[ upsellProductSlug ] ) {
```

Comment:
I feel that the check for a `yearly` product is quite obfuscated here. Do you think it's feasible and would make sense to abstract this check into a separate method like `isYearlyProduct()` (and possibly adding a `isMonthlyProduct()` counterpart)?